cypress/integration/seededFlows/loggedOutFlows/namespacedForemMobile.spec.js

```diff
@@ -0,0 +1,83 @@
+describe('Namespaced ForemMobile functions', () => {
+  // Make the runtime act as ForemWebView context
+  const runtimeStub = {
+    onBeforeLoad: (win) => {
+      Object.defineProperty(win.navigator, 'userAgent', {
+        value:
+          'Mozilla/5.0 (iPhone; CPU iPhone OS 14_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 ForemWebView/1.0',
+      });
+      Object.defineProperty(win.navigator, 'platform', { value: 'iPhone' });
+    },
+  };
+
+  beforeEach(() => cy.testSetup());
+
+  describe('within ForemWebView context', () => {
+    beforeEach(() => cy.visit('/', runtimeStub));
+
+    it('should load the ForemMobile namespaced functions', () => {
+      cy.window().then((win) => {
+        assert.isNumber(win.ForemMobile.retryDelayMs);
+        assert.isFunction(win.ForemMobile.getUserData);
+        assert.isFunction(win.ForemMobile.getInstanceMetadata);
+        assert.isFunction(win.ForemMobile.registerDeviceToken);
+        assert.isFunction(win.ForemMobile.unregisterDeviceToken);
+        assert.isFunction(win.ForemMobile.injectJSMessage);
+        assert.isFunction(win.ForemMobile.injectNativeMessage);
+        assert.isFunction(win.ForemMobile.userSessionBroadcast);
+      });
+    });
+
+    it('should return the instance metadata JSON when requested', () => {
+      cy.window().then((win) => {
+        var res = JSON.parse(win.ForemMobile.getInstanceMetadata());
+        assert.isObject(res);
+        assert.equal(res.domain, 'localhost:3000');
+        assert.isString(res.logo);
+        assert.isString(res.name);
+      });
+    });
+
+    it('should return empty user data when logged out', () => {
+      cy.window().then((win) => {
+        assert.isUndefined(win.ForemMobile.getUserData());
+      });
+    });
+
+    it('should return user data JSON when logged in', () => {
+      // Attempt to login user and fetch its data
+      cy.fixture('users/adminUser.json').as('user');
+      cy.get('@user').then((user) => {
+        cy.loginUser(user).then(() => {
+          cy.visit('/settings', runtimeStub)
+            .then(() => cy.window())
+            .then((win) => {
+              var res = JSON.parse(win.ForemMobile.getUserData());
+              assert.isObject(res);
+            });
+        });
```

Comment:
[This spec is failing](https://app.travis-ci.com/github/forem/forem/jobs/547412785#L2212) because the user isn't signed in. I'm not too sure what's wrong with this 🤔 

The `cy.visit('/settings', runtimeStub)` is redirecting back to `/enter`

<img width="1090" alt="Screen Shot 2021-11-09 at 16 37 30" src="https://user-images.githubusercontent.com/6045239/141016682-57821671-9fe1-4026-92a3-b3c7705c5841.png">

Despite the sign in request did return a status code `200`

<img width="1427" alt="Screen Shot 2021-11-09 at 16 36 49" src="https://user-images.githubusercontent.com/6045239/141016779-a76b292a-f008-4449-a360-6cc886f6647b.png">